src/brsTypes/components/RoAppInfo.ts

```diff
@@ -0,0 +1,141 @@
+import { BrsBoolean, BrsString, BrsValue, ValueKind } from "../BrsType";
+import { BrsComponent } from "./BrsComponent";
+import { BrsType } from "..";
+import { Callable, StdlibArgument } from "../Callable";
+import { Interpreter } from "../../interpreter";
+import * as PP from "../../preprocessor";
+
+export class RoAppInfo extends BrsComponent implements BrsValue {
+    readonly kind = ValueKind.Object;
+
+    private static _manifest = PP.getManifestSync(process.cwd());
+
+    constructor() {
+        super("roAppInfo");
+
+        this.registerMethods({
+            ifAppInfo: [
+                this.getID,
+                this.isDev,
+                this.getVersion,
+                this.getTitle,
+                this.getSubtitle,
+                this.getDevID,
+                this.getValue,
+            ],
+        });
+    }
+
+    toString(parent?: BrsType) {
+        return "<Component: roAppInfo>";
+    }
+
+    equalTo(other: BrsType) {
+        return BrsBoolean.False;
+    }
+
+    /**
+     * Originally returns the app's channel ID or 'dev' for sideloaded applications.
+     * @returns {string} - 'dev'
+     */
+    private getID = new Callable("getID", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (interpreter: Interpreter) => {
+            return new BrsString("dev");
+        },
+    });
+
+    /**
+     * Returns true if the application is sideloaded, i.e. the channel ID is "dev".
+     * @returns {boolean} - true
+     */
+    private isDev = new Callable("isDev", {
+        signature: {
+            args: [],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter) => {
+            return BrsBoolean.True;
+        },
+    });
+
+    private getVersion = new Callable("getVersion", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (interpreter: Interpreter) => {
+            let manifest = RoAppInfo._manifest;
+
+            let version = ["major_version", "minor_version", "build_version"]
+                .map((key) => manifest.get(key))
+                .filter((key) => !!key)
+                .join(".");
+            version = version !== "" ? version : "..";
```

Comment:
I feel like returning `0.0.0` instead of `..` (if no values are provided in manifest) it's a bit more meaningful, what do you think?

Reply:
Does that mean including no `build_version` would return `1.2.`?

Reply:
Oh wow, didn't know that. Let's keep things aligned with Roku then